docs/secure-communication-agents.asciidoc

```diff
@@ -1,21 +1,213 @@
 [[secure-communication-agents]]
 == Secure communication with APM Agents
 
-To secure the communication between APM Agents and the APM Server:
+There are two ways to secure the communication between APM Agents and the APM Server.
+Both options can be enabled in parallel,
+allowing Elastic APM agents to chose whichever mechanism they support.
 
-. <<secret-token,Set a secret token in your Agents and Server>>
-. <<ssl-setup,Enable SSL/TLS in APM Server>>
-. <<https-in-agents,Enable HTTPS in your APM Agents>>
+As soon as secure communication is enabled, requests without a valid token will be denied by APM Server.
+As RUM endpoints cannot be secured, they are exempt from this rule.
 
-[[secret-token]]
+* <<api-key,API keys>>
+* <<secret-token,Secret token>>
+
+[[api-key]]
+=== API Keys
+
+//**************************************************
+// NEW STUFF
+//**************************************************
+
+You can configure an API key to authorize requests to the APM Server.
+When both the agents and the APM servers are configured with the same API key,
```

Comment:
Only configure in apm-server for self instrumentation for the apm go agent. Create in server and set in agent. Configure isn't the right word.